tensorflow/core/framework/kernel_def.proto

```diff
@@ -16,6 +16,12 @@ message KernelDef {
 
   // Type of device this kernel runs on.
   string device_type = 2;
+ 
+  // Sub-type of device this kernel runs on.
```

Comment:
This comment needs to describe what a sub-device means.  A formal definition is not necessary, some examples would be sufficient.

Reply:
done

Reply:
`X_GPU` does not shine much light on the matter. :)

I meant something like "For instance, the DirectML TensorFlow backend has a subdevice type that is backed by ... (presumably DirectML APIs)."